yarn-project/archiver/src/archiver/lmdb_archiver_store.ts

```diff
@@ -163,15 +165,10 @@ export class LMDBArchiverStore implements ArchiverDataStore {
         .getRange(this.#computeBlockRange(start, limit))
         .filter(({ value }) => value.block)
         .map(({ value }) => {
-          const block = L2Block.fromBuffer(asBuffer(value.block!));
-          if (value.encryptedLogs) {
-            block.attachLogs(L2BlockL2Logs.fromBuffer(asBuffer(value.encryptedLogs)), LogType.ENCRYPTED);
-          }
-
-          if (value.unencryptedLogs) {
-            block.attachLogs(L2BlockL2Logs.fromBuffer(asBuffer(value.unencryptedLogs)), LogType.UNENCRYPTED);
-          }
-
+          const block = L2Block.fromBuffer(
+            asBuffer(value.block!),
+            value.blockHash ? asBuffer(value.blockHash) : undefined,
+          );
```

Comment:
Not having an interface for L2Block is strange to me since it seems a major point of interaction for clients. Having two separate interfaces for blocks with/without logs feels like it would get painful as well. Maybe just:
```typescript
interface L2Block {
    ...
    newEncryptedLogs?: L2BlockL2Logs,
    ...
}
type L2BlockWithLogs = L2Block & { newEncryptedLogs: L2BlockL2Logs };
type L2BlockWithoutLogs = L2Block & { newEncryptedLogs: undefined };
```
then in the class implementation of `L2Block`. 
```typescript
hasLogs(): this is L2BlockWithLogs {
    return this.newEncryptedLogs !== undefined
}
```

Reply:
I agree with both of you, but changes are coming to this section of the code to bring it inline with the Yello paper and I'd prefer to wait for them to land before touching it more than needed if that's alright :)